### PR TITLE
demonstrate `format` and `as_cmdstan_fit()` in vignettes

### PR DESCRIPTION
**Pull request by jgabry**
#### Submission Checklist

- [x] Run unit tests
- [x] Declare copyright holder and agree to license (see below)

#### Summary

* demonstrate `format` argument to `draws()`
* demonstrate `as_cmdstan_fit()`
* other tiny edits

#### Copyright and Licensing

Please list the copyright holder for the work you are submitting 
(this will be you or your assignee, such as a university or company): 
**Columbia University**

By submitting this pull request, the copyright holder is agreeing to 
license the submitted work under the following licenses:

- Code: BSD 3-clause (https://opensource.org/licenses/BSD-3-Clause)    
- Documentation: CC-BY 4.0 (https://creativecommons.org/licenses/by/4.0/)

### CODE REVIEW

File: vignettes/cmdstanr.Rmd [hunk 1]
```diff
@@ -194,23 +194,35 @@ fit$summary("theta", pr_lt_half = ~ mean(. <= 0.5))
...
+draws_df <- fit$draws(format = "df")
+str(draws_df)
 print(draws_df)
 ```
 
+To convert an existing draws object to a different format use the
+`posterior::as_draws_*()` functions.
+
+```{r as_draws}
+# this should be identical to draws_df created via draws(format = "df")
```
Comment (rok-cesnovar) [inline vignettes/cmdstanr.Rmd:218]: Maybe mention that subsequent converts will be slower than just setting the appropriate format in the draws() call. It wont matter in most cases though.

**Review by rok-cesnovar — COMMENTED** [review]
1 minor comment, otherwise looks good. Do we need to rebuild vignettes?

File: vignettes/cmdstanr.Rmd (hunk 1 above)
Comment (jgabry) [reply vignettes/cmdstanr.Rmd:218]: Good call, but I guess it kind of depends on which option we decide on for solving #523 right?

Comment (rok-cesnovar) [reply vignettes/cmdstanr.Rmd:218]: Yeah, if we decide to convert on return for draws matrix it will be the same for draws matrix.

Comment (jgabry) [reply vignettes/cmdstanr.Rmd:221]: > Maybe mention that subsequent converts will be slower than just setting the appropriate format in the draws() call. It wont matter in most cases though.

Coming back to this now finally. Just added this: 

> In general, converting to a different draws format in this way will be slower
than just setting the appropriate format initially in the call to the `$draws()`
method, but in most cases the speed difference will be minor.

**Comment by jgabry** [comment]
I realized I never finished this. I just pushed a few more edits (and addressed your review comment) and rebuilt the affected vignettes.

**Review by rok-cesnovar — APPROVED** [review]
Looks great!

**Pull request merged**